tutorials/text/deduplication/semantic/semantic_step_by_step.ipynb

```diff
@@ -0,0 +1,1114 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "54c8fe16",
+   "metadata": {},
+   "source": [
+    "# Semantic Deduplication on Text\n",
+    "\n",
+    "GPU accelerated implementation of [SemDeDup: Data-efficient learning at web-scale through semantic deduplication](https://arxiv.org/abs/2303.09540)  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8a2c97d2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "os.environ[\"LOGURU_LEVEL\"] = \"ERROR\"\n",
+    "\n",
+    "import pandas as pd\n",
+    "\n",
+    "input_path_to_data = \"/path/to/data\"\n",
+    "semantic_out_dir = os.path.abspath(\"./output/step_by_step\")\n",
+    "output_path = os.path.join(semantic_out_dir, \"output\")\n",
+    "cache_path = os.path.join(semantic_out_dir, \"cache\")\n",
+    "# this can be either of jsonl or parquet\n",
+    "# #however if you change it to json you'll need to change embedding generations reader to also be JsonlReader\n",
+    "input_filetype = \"parquet\"\n",
+    "output_filetype = \"parquet\"  # this can be either of jsonl or parquet"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9f9e2c0f",
+   "metadata": {},
+   "source": [
+    "# Running SemDedup workflow by workflow\n",
+    "\n",
+    "Here we intentionally break it down into executing three workflows. \n",
```

Comment:
Should we mention this can be useful for Slurm jobs, etc.?

Reply:
Good callout!